lib/list.gd

```diff
@@ -170,7 +170,7 @@ InstallTrueMethod(HasLength,IsPlistRep);
 ##  <#/GAPDoc>
 ##
 DeclareOperationKernel( "IsBound[]",
-    [ IsList, IsObject ],
+    [ IsList, IS_INT ],
```

Comment:
FYI, this change simply reverts things to how they were before 1485ae9ef8bd7025e7e6da444328335c561a6bc1

Also, `IsPosInt` is defined in `lib/cyclotom.g`, and I just verified `IsPosInt` is not yet defined when `list.gd` is loadd.